BDArmory/Modules/BDModulePilotAI.cs

```diff
@@ -1943,6 +1961,85 @@ void FlyOrbit(FlightCtrlState s, Vector3d centerGPS, float radius, float speed,
             FlyToPosition(s, targetPosition);
         }
 
+        public void ClearWaypoints()
+        {
+            Debug.Log("[BDArmory.BDModulePilotAI] Cleared waypoints");
+            this.waypoints = null;
+            this.activeWaypointIndex = -1;
+        }
+        public void SetWaypoints(List<Vector3> waypoints)
+        {
+            if( waypoints == null || waypoints.Count == 0 )
+            {
+                this.activeWaypointIndex = -1;
+                this.waypoints = null;
+                this.waypointScores = null;
+                return;
+            }
+            Debug.Log(string.Format("[BDArmory.BDModulePilotAI] Set {0} waypoints", waypoints.Count));
+            this.waypoints = waypoints;
+            this.waypointScores = new List<float>();
+            this.activeWaypointIndex = 0;
+        }
+        public bool IsFlyingWaypoints()
+        {
+            return this.activeWaypointIndex >= 0 && waypoints != null && waypoints.Count > 0;
+        }
+        public List<float> GetWaypointScores()
+        {
+            return this.waypointScores;
+        }
+
+        private double lastRange = 0;
+        private double dRange = 0;
+        void FlyWaypoints(FlightCtrlState s)
```

Comment:
This is the actual waypoint following code. It doesn't appear to work with guard mode.